cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -84,7 +84,12 @@ type Etcd struct {
 type NodeConfiguration struct {
 	metav1.TypeMeta
 
-	Discovery Discovery
+	Discovery         Discovery
```

Comment:
I am removing the --discovery flag for the `join` command in this PR.  There is still --discovery for `init`. I will change that in a later PR.